src/components/show-products/index.js

```diff
@@ -10,16 +11,52 @@ import {
 import { TimeFrames } from '../../utils/timeFrames';
 import { compareDates } from '../../utils/compareDates';
 import ContentContainer from '../content-container';
+import { doc, setDoc, serverTimestamp } from '@firebase/firestore';
+import { db } from '../../lib/firebase';
+
 import './styles.css';
 
 export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
   const timeFrames = TimeFrames;
+  const ONE_DAY = 1;
 
-  const handleOnChange = (productID) => {
+  const handleOnChange = (e, productID) => {
     updatePurchaseDate(productID, storedValue);
+    const item = products?.find((product) => product.id === productID);
+    const daysSinceLastTransaction =
+      item.createdAt.toDate() !== 0 ? compareDates(item.createdAt.toDate()) : 0;
+    const checked = e.target.checked;
+
+    const estimatedTime = calculateEstimate(
+      parseInt(item.timeFrame),
+      daysSinceLastTransaction,
+      item.numberOfPurchases,
+    );
+    if (checked) {
+      const itemRef = doc(db, storedValue, productID);
+      setDoc(
+        itemRef,
+        {
+          lastPurchaseDate: serverTimestamp(),
+          daysUntilNextPurchase: estimatedTime,
+          numberOfPurchases: item.numberOfPurchases + 1,
+        },
+        { merge: true },
+      );
+    } else {
+      const itemRef = doc(db, storedValue, productID);
+      setDoc(
+        itemRef,
+        {
+          lastPurchaseDate: null,
+          numberOfPurchases: item.numberOfPurchases - 1,
+        },
+        { merge: true },
+      );
+    }
```

Comment:
It's interesting. Also, I'm surprised 😮  about lines 49 - 59 (else - block).

Why? Because...
1. I don't think it's the behavior expected (**you shouldn't check and uncheck** immediately and manually **_when you want to do it._** Your checkbox should be **disabled** 👀 ).
2. This checkbox should be "enabled" after 24-hours.

https://user-images.githubusercontent.com/8689897/140453922-b6357140-d8d8-4070-b751-347b81d7b883.mp4

Of course, this is my opinion. Feel free to ask in our channel for more opinions or thoughts. 👍

Reply:
☝️ This is totally related to my previous comment. Probably we don't want to remove the last purchase date for these cases.